app/Http/Controllers/BookingAPIController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Booking;
+use App\Models\Room;
+use Carbon\CarbonPeriod;
+use http\Env\Response;
+use Illuminate\Http\Request;
+
+class BookingAPIController extends Controller
+{
+    public function create(Request $request)
+    {
+
+        $request->validate([
+            'room_id' => 'required|integer|exists:rooms,id',
+            'customer' => 'required|string|max:255',
+            'guests' => 'required|integer|min:1',
+            'start' => 'required|date|date_format:Y-m-d|after_or_equal:today',
+            'end' => 'required|date|date_format:Y-m-d|after:start',
+        ]);
+
+        $room = Room::find($request->room_id);
+        $startDate = $request->start;
+        $endDate = $request->end;
+
+
+        $bookingConflict = Booking::where('room_id', $room->id)
+            ->where(function ($query) use ($startDate, $endDate) {
+                $query->whereBetween('start', [$startDate, $endDate])
+                    ->orWhereBetween('end', [$startDate, $endDate])
+                    ->orWhere(function ($query) use ($startDate, $endDate) {
+                        $query->where('start', '<=', $startDate)
+                            ->where('end', '>=', $endDate);
+                    });
+            })
+            ->exists();
```

Comment:
This would be really nice to have as it's own service. 

Partly because it hides the complicated query behind a method, and also because this may well be something that needs to be repeated in the future (although probably not for these stories)

Reply:
Would suggest creating a service called `BookingService` (in app/Services - you need to make that folder). Never know what other stuff you might end up putting in it.